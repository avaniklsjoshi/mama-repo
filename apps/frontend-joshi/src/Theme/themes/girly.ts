export default {
  themeType: "girly",
  colors: {
    background: "#f1b7ea",
    text: "#fff"
  },
  header: {
    background: "#000",
    text: "#fff"
  },
  footer: {
    background: "#262726",
    text: "#fff"
  },
  section: {
    background: "#008F7A",
    text: "#fff"
  }
};
