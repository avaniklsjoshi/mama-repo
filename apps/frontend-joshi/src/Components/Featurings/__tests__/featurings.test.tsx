/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
import { act, fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import girly from "../../../Theme/themes/girly";
import Layout from "../layout";
import Menu from "../menu";
import WelcomePage from "../welcomePage";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: any) => key
  })
}));

describe("Features Page", () => {
  it("Featurings: renders Welcome Page", () => {
    const msg = "Love you Joshi!";
    const { container } = render(<WelcomePage welcomeMsg={msg} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText(msg)).toBeInTheDocument();
  });
  it("Featurings: renders Features Page", () => {
    const mock = jest.fn();
    const { container } = render(
      <ThemeProvider theme={girly}>
        <BrowserRouter>
          <Menu activeComponent={mock} routeName="uiLearnings" />
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    const menuItem1 = screen.getAllByText(/Parallax Effect/i);
    expect(menuItem1[0]).toBeInTheDocument();
    const menuItem2 = screen.getAllByText(/Bubble Effect/i);
    expect(menuItem2[0]).toBeInTheDocument();
    const menuItem3 = screen.getAllByText(/Designer Title/i);
    expect(menuItem3[0]).toBeInTheDocument();
  });
  it("Featurings: renders Layout", () => {
    const SubRouteChildComponent = "Hi Avani!";
    const { container } = render(
      <ThemeProvider theme={girly}>
        <BrowserRouter>
          <Layout routeName="home">
            <div>
              <h1>{SubRouteChildComponent}</h1>
            </div>
          </Layout>
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText(SubRouteChildComponent)).toBeInTheDocument();
  });
});
