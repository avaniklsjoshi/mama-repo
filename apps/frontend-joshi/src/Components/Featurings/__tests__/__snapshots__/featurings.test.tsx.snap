// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Page Featurings: renders Features Page 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="left-menu-box MuiBox-root css-0"
    style="text-align: initial; height: 100%;"
  >
    <div
      class="PrivateHiddenCss-root PrivateHiddenCss-smUp css-od3h8q-PrivateHiddenCss-root"
    />
    <div
      class="PrivateHiddenCss-root PrivateHiddenCss-xsDown css-1f8mm7w-PrivateHiddenCss-root"
    >
      <div
        class="MuiDrawer-root MuiDrawer-docked left-drawer css-sc4n0q-MuiDrawer-docked"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            style="align-self: baseline; margin: 0px; padding: 0px;"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters menu-item css-1nxmd3h-MuiListItem-root"
              data-testid="menu-click"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Parallax Effect
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters menu-item css-1nxmd3h-MuiListItem-root"
              data-testid="menu-click"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Bubble Effect
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters menu-item css-1nxmd3h-MuiListItem-root"
              data-testid="menu-click"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Designer Title
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Features Page Featurings: renders Layout 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="sc-bczRLJ dxuRWx"
  >
    <div
      class="sc-gsnTZi eLnOdR"
    >
      <a
        class="back-link"
        data-testid="back-link"
        style="color: red;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowBackIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
        back
      </a>
    </div>
    <div
      class="sub-route-container"
    >
      <div
        class="left-bar"
      >
        <div
          class="left-menu-box MuiBox-root css-0"
          style="text-align: initial; height: 100%;"
        >
          <div
            class="PrivateHiddenCss-root PrivateHiddenCss-smUp css-od3h8q-PrivateHiddenCss-root"
          />
          <div
            class="PrivateHiddenCss-root PrivateHiddenCss-xsDown css-1f8mm7w-PrivateHiddenCss-root"
          >
            <div
              class="MuiDrawer-root MuiDrawer-docked left-drawer css-sc4n0q-MuiDrawer-docked"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  style="align-self: baseline; margin: 0px; padding: 0px;"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ChevronLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <ul
                  class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="right-bar"
      >
        <div
          class="right-bar-title"
        />
        <div>
          <h1>
            Hi Avani!
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Features Page Featurings: renders Welcome Page 1`] = `
<div>
  <div>
    Love you Joshi!
  </div>
</div>
`;
